x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/http/sender/RequestExecutorService.java

```diff
@@ -174,6 +176,10 @@ public int queueSize() {
         return rateLimitGroupings.values().stream().mapToInt(RateLimitingEndpointHandler::queueSize).sum();
     }
 
+    public Collection<RateLimitingEndpointHandler> rateLimitingEndpointHandlers() {
```

Comment:
Just to reiterate, hopefully we can implement this without using any getters in this class.